utils/s3/client.hh

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/file.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/http/client.hh>
+
+using namespace seastar;
+
+namespace s3 {
+
+struct range {
+    uint64_t off;
+    size_t len;
+};
+
+class client : public enable_shared_from_this<client> {
+    class upload_sink;
+    class readable_file;
+    sstring _host;
+    http::experimental::client _http;
+
+    struct private_tag {};
+    future<std::vector<temporary_buffer<char>>> get_object(sstring object_name, std::optional<std::pair<uint64_t, size_t>> range);
```

Comment:
std::optional<range> range?

Reply:
Yikes, it's a trash forward-declaration, need to remove one